addons/website_sale_stock/models/sale_order.py

```diff
@@ -8,20 +8,16 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
-    def _get_warehouse_available(self):
-        self.ensure_one()
-        warehouse = self.website_id._get_warehouse_available()
-        if not warehouse and self.user_id and self.company_id:
-            warehouse = self.user_id.with_company(self.company_id.id)._get_default_warehouse_id()
-        if not warehouse:
-            warehouse = self.env.user._get_default_warehouse_id()
-        return warehouse
-
     def _compute_warehouse_id(self):
         website_orders = self.filtered('website_id')
         super(SaleOrder, self - website_orders)._compute_warehouse_id()
         for order in website_orders:
-            order.warehouse_id = order._get_warehouse_available()
+            if order.website_id.warehouse_id:
+                order.warehouse_id = order.website_id.warehouse_id
+            else:
+                super(SaleOrder, order)._compute_warehouse_id()
+            if not order.warehouse_id:
+                order.warehouse_id = self.env.user._get_default_warehouse_id()
```

Comment:
If I understand correctly, the SO's warehouse will still be the same as before, but we compute it via a different codepath (unlike the website's warehouse which won't have a fallback anymore), right?

Reply:
yes, for the website the user should set it explicitly now, and for SO nothing should change.